sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -417,6 +417,14 @@ object NullPropagation extends Rule[LogicalPlan] {
         case left :: Literal(null, _) :: Nil => Literal.create(null, e.dataType)
         case _ => e
       }
+
+      // If the value expression is NULL then transform the In expression to
+      // Literal(null)
+      case e @ In(v, list) => v match {
```

Comment:
we can just write `case In(Literal(null, _), list) => Literal.create(null, BooleanType)`

We have `ConstantFolding` rule which will turn the `v` into literal if it's foldable.